src/modules/condition/failed.rs

```diff
@@ -80,23 +80,27 @@ impl TranslateModule for Failed {
             if self.is_question_mark {
                 // if the failed expression is in the main block we need to clear the return value
                 let clear_return = if !self.is_main {
-                    let (name, id, variant) = meta.fun_name.clone().expect("Function name not set");
-                    format!("__AF_{name}{id}_v{variant}=''")
+                    let fun_name = meta.fun_name.as_ref().expect("Function name not set");
+                    format!("{}={}", fun_name.mangled_name(), fun_name.default_value())
```

Comment:
This fixes a bug where failing functions would return the previous array value with the first item blanked out, because the default return value was Bash string `''` not Bash array `()`.  Try running the following Amber script with commits up to "5212ef5 Add glob functions to standard library":
```
hwalters@Ghostwheel ~/git/amber (HEAD) 
$ cat glob.ab
#!/usr/bin/env amber
import * from "std/fs"

echo "Calling glob(\"*.md\")"
loop file in trust glob("*.md") {
    echo "Found \"{file}\""
}

echo "Calling glob(\"*.xyz\")"
loop file in trust glob("*.xyz") {
    echo "Found \"{file}\""
}
```
```
hwalters@Ghostwheel ~/git/amber (HEAD) 
$ ./glob.ab
Calling glob("*.md")
Found "CONTRIBUTING.md"
Found "LICENSE.md"
Found "NIX.md"
Found "README.md"
Calling glob("*.xyz")
Found ""
Found "LICENSE.md"
Found "NIX.md"
Found "README.md"
```
This is what it does with the latest commit:
```
hwalters@Ghostwheel ~/git/amber (implement-stdlib-glob-function) 
$ ./glob.ab
Calling glob("*.md")
Found "CONTRIBUTING.md"
Found "LICENSE.md"
Found "NIX.md"
Found "README.md"
Calling glob("*.xyz")
```